nodeadm/internal/api/status.go

```diff
@@ -0,0 +1,66 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go/aws"
+	ec2extra "github.com/awslabs/amazon-eks-ami/nodeadm/internal/aws/ec2"
+)
+
+// Fetch information about the ec2 instance using IMDS data.
+// This information is stored into the internal config to avoid redundant calls
+// to IMDS when looking for instance metadata
+func GetInstanceDetails(ctx context.Context, imdsClient *imds.Client, ec2Client *ec2.Client) (*InstanceDetails, error) {
+	instanceIdenitityDocument, err := imdsClient.GetInstanceIdentityDocument(ctx, &imds.GetInstanceIdentityDocumentInput{})
+	if err != nil {
+		return nil, err
+	}
+
+	macResponse, err := imdsClient.GetMetadata(ctx, &imds.GetMetadataInput{Path: "mac"})
+	if err != nil {
+		return nil, err
+	}
+	mac, err := io.ReadAll(macResponse.Content)
+	if err != nil {
+		return nil, err
+	}
+
+	privateDNSName, err := getPrivateDNSName(ec2Client, instanceIdenitityDocument.InstanceID)
```

Comment:
whats the approach to work around this call for e2e tests?
do we need a general aws service mock or maybe an env-var escape hatch?

Reply:
Yeah...need to figure that out. I don't want to mock out the EC2 API, but it might be worth laying the groundwork now